src/Network/Minio/Data.hs

```diff
@@ -238,24 +272,33 @@ data UploadInfo = UploadInfo {
 data ListObjectsResult = ListObjectsResult {
     lorHasMore   :: Bool
   , lorNextToken :: Maybe Text
-  , lorObjects   :: [ObjectInfo]
+  , lorObjects   :: [ContentInfo]
   , lorCPrefixes :: [Text]
   } deriving (Show, Eq)
 
 -- | Represents result from a listing of objects version 1 in a bucket.
 data ListObjectsV1Result = ListObjectsV1Result {
     lorHasMore'   :: Bool
   , lorNextMarker :: Maybe Text
-  , lorObjects'   :: [ObjectInfo]
+  , lorObjects'   :: [ContentInfo]
   , lorCPrefixes' :: [Text]
   } deriving (Show, Eq)
 
+-- | Represents information about an object in list objects response.
+data ContentInfo = ContentInfo {
```

Comment:
We should have one type (say) `ObjectInfo`. We should __not__ export `ObjectInfo` constructor. This is a breaking change like the introduction of GetObjectOptions (which is fine). We should export accessors like `oiObject`, `oiModTime`, `oiETag`, `oiSize` and `oiMetadata`. `oiMetadata` should return Map.empty when called on a listObjects result element. It should return the user-metadata saved with the object as a Map of header name and value.

Reply:
The problem occurs with usage of uncurry4 and zip4 in XmlParser.hs, I couldnt find an easy way to do

```
etags = etagList ++ repeat ""
```

I tried like 

```
metadatas = etagList ++ repeat []
```

But didn't work and ended but the way it is now. Let me know i am happy to change it back  . I deliberately did it so that we can discuss :P.

Also I didn't want to keep it as Map, but have a much simpler form such as `[(Text, Text)]` . Do you not think it's better ?

Reply:
Actually used your approach and @donatello helped on understanding the uncurry, zip behavior.